sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/HadoopFsRelation.scala

```diff
@@ -71,7 +73,13 @@ case class HadoopFsRelation(
 
   override def sizeInBytes: Long = {
     val compressionFactor = sqlContext.conf.fileCompressionFactor
-    (location.sizeInBytes * compressionFactor).toLong
+    val defaultSize = (location.sizeInBytes * compressionFactor).toLong
+    location match {
+      case _: CatalogFileIndex if sparkSession.sessionState.conf.fallBackToHdfsForStatsEnabled =>
+        DDLUtils.sizeInBytesFallBackToHdfs(sparkSession,
+          new Path(location.asInstanceOf[CatalogFileIndex].table.location), defaultSize)
```

Comment:
if you name the case match variable, you already have it as a cast here. But yeah this is cleaner.

Reply:
Yes. updated.

Reply:
Yes. Updated.